app/assets/javascripts/3dbio_viewer/src/webapp/components/protvista/Tooltip.tsx

```diff
@@ -32,40 +38,16 @@ export const Tooltip: React.FC<TooltipProps> = React.memo(props => {
     //no react.memo as is finally rendered to string
     const nmrSource = getSource(sources, "NMR");
     const score = alignmentScore ? alignmentScore + " %" : undefined; // aligmentScore is never being set on code
-    const isStructureCoverage = subtrack.accession === trackDefinitions.structureCoverage.id;
     const isNMR = subtrack.accession === "nmr";
 
-    const isCovered =
-        isStructureCoverage ||
-        alignment.some(
-            interval => fragment.start >= interval.start && fragment.end <= interval.end
-        );
+    const covered = isCovered(alignment, fragment);
+    const partiallyCovered = isPartiallyCovered(alignment, fragment);
+    const notCovered = isNotCovered(alignment, fragment);
 
-    const isPartiallyCovered = alignment.some(
-        interval => fragment.start <= interval.end && fragment.end >= interval.start
-    );
-
-    const isNotCovered = alignment.every(
-        interval => fragment.start > interval.end || fragment.end < interval.start
-    );
-
-    const coverage = isNotCovered
-        ? {
-            value: i18n.t("Not in Structure Coverage"),
-            className: "error",
-            title: i18n.t(
-                "This annotation is part of the full UniProt sequence but lies outside the region captured in the 3D structure (PDB) for this specific chain. It may correspond to a flexible, disordered, or missing part of the protein that was not resolved during structure determination."
-            ),
-        }
-        : isPartiallyCovered && !isCovered
-            ? {
-                value: i18n.t("Partially in Structure Coverage"),
-                className: "warning",
-                title: i18n.t(
-                    "This annotation is partially covered by the 3D structure (PDB) for this specific chain. The structure may capture part of this region, but additional functional or structural details extend beyond the resolved area."
-                ),
-            }
-            : undefined;
+    const coverage = [
+        { condition: notCovered, details: coverageDetails.notCovered },
+        { condition: partiallyCovered && !covered, details: coverageDetails.partiallyCovered },
+    ].find(item => item.condition)?.details;
```

Comment:
As discussed, this way of writing conditionals is also very declarative. Just note that all the return values are calculated (when using if blocks, they are not unless it enters the branch). Here is not a problem. When that's a problem, instead of the value, put a function in `details`

Reply:
Good point!! 🚀